wandb/apis/reports/v2/interface.py

```diff
@@ -1470,20 +1469,22 @@ def from_model(cls, model: internal.ReportViewspec):
         if blocks[-1] == internal.Paragraph():
             blocks = blocks[:-1]
 
-        return cls(
+        obj = cls(
```

Comment:
The fields added later have `init=False`, which means they are not part of the constructor -- if you try to do `wr.Report(id=...)` you'll get an error.

They are intentionally not part of the constructor because we don't want users manually setting the ID, or messing with internal parts of the report like `_authors`, `_ref`, etc.

Those attrs are still important parts of the report though, so we add them back as private attrs.

In the case of a brand new report created using the normal `__init__` constructor, these fields would be blank (e.g. the report ID does not exist yet because it has not been created server-side).

Reply:
Ah, that makes sense to me now. Thanks!